src/frontend/admin/src/components/templates/enrollments/inputs/RHFSelectEnrollmentState.tsx

```diff
@@ -20,26 +20,16 @@ const messages = defineMessages({
 export function RHFSelectEnrollmentState(props: RHFSelectProps) {
   const intl = useIntl();
   const options: SelectOption[] = useMemo(() => {
-    const result: SelectOption[] = [];
-    Object.values(EnrollmentState).forEach((v) => {
-      if (v) {
-        result.push({
-          label: intl.formatMessage(
-            enrollmentStateMessages[v as EnrollmentState],
-          ),
-          value: v + "",
-        });
-      }
-    });
-    return result;
+    return Object.values(EnrollmentState).map((v) => ({
+      label: intl.formatMessage(enrollmentStateMessages[v]),
+      value: v,
+    }));
   }, []);
 
   return (
     <RHFSelect
       getOptionLabel={(value) =>
```

Comment:
I think it's better and less verbose than enforce value through `as` below 
```suggestion
      getOptionLabel={(value: EnrollmentState) =>
```